### PR TITLE
Skwasm Font Loading

### PR DESCRIPTION
**Pull request by eyebrowsoffire**
This implements font loading for the skwasm renderer.

In addition, it does some pretty major refactors:
1) Simplified the font collection interface to just have a single `loadAssetFonts` call, without the separate registration and debug fonts loading stuff
2) Debug fonts load now through http/asset mocking mechanisms instead of having a separate `downloadDebugTestFonts` call
3) Consolidated a few of our different unit test setup functions into a single `setUpUnitTests` function

### CODE REVIEW

File: lib/web_ui/lib/src/engine/fonts.dart [hunk 1]
```diff
@@ -3,32 +3,126 @@
...
+  FontAsset(this.asset, this.descriptors);
+
+  String asset;
+  Map<String, String> descriptors;
+}
+
+class FontFamily {
+  FontFamily(this.name, this.fontAssets);
+
+  String name;
```
Comment (kevmoo) [inline lib/web_ui/lib/src/engine/fonts.dart:22]:
```suggestion
  final String name;
```

File: lib/web_ui/lib/src/engine/fonts.dart [hunk 2]
```diff
@@ -3,32 +3,126 @@
...
+  List<FontAsset> fontAssets;
+}
+
+class FontManifest {
+  FontManifest(this.families);
+
+  List<FontFamily> families;
```
Comment (kevmoo) [inline lib/web_ui/lib/src/engine/fonts.dart:29]:
```suggestion
  final List<FontFamily> families;
```

File: lib/web_ui/lib/src/engine/fonts.dart [hunk 3]
```diff
@@ -3,32 +3,126 @@
...
 import 'dart:typed_data';
 
-import 'assets.dart';
+import 'package:ui/src/engine.dart';
+
+class FontAsset {
+  FontAsset(this.asset, this.descriptors);
+
+  String asset;
+  Map<String, String> descriptors;
```
Comment (kevmoo) [inline lib/web_ui/lib/src/engine/fonts.dart:16]:
```suggestion
  final String asset;
  final Map<String, String> descriptors;
```

File: lib/web_ui/lib/src/engine/skwasm/skwasm_impl/font_collection.dart [hunk 4]
```diff
@@ -3,34 +3,116 @@
...
 
 import 'dart:typed_data';
 
 import 'package:ui/src/engine.dart';
+import 'package:ui/src/engine/skwasm/skwasm_impl.dart';
 
 class SkwasmFontCollection implements FlutterFontCollection {
+  SkwasmFontCollection() : _handle = fontCollectionCreate();
+
+  FontCollectionHandle _handle;
```
Comment (kevmoo) [inline lib/web_ui/lib/src/engine/skwasm/skwasm_impl/font_collection.dart:18]: final?

Comment (eyebrowsoffire) [reply lib/web_ui/lib/src/engine/skwasm/skwasm_impl/font_collection.dart:18]: `clear` recreates the underlying font collection, so the handle can't be final.

**Pull request merged**